packages/@react-stately/grid/src/GridCollection.ts

```diff
@@ -33,6 +33,9 @@ export class GridCollection<T> implements IGridCollection<T> {
       // we can skip this node and its children. We always visit columns though,
       // because we depend on order to build the columns array.
       let prevNode = this.keyMap.get(node.key);
+      if (nodePopulate) {
+        node = nodePopulate(node);
```

Comment:
this function seemed like the best way to add info for each grid cell since we're trying to not have `TableCollection` go over the cells, but if there's a better way, definitely let me know